Panic on Ctrl-C to ipfs add

<!--
Output From `ipfs version --all`

If your version is older than *0.4.11* please update
and re-check if the problem persists.
-->
#### Version information: 0.4.14-dev-b7824ba (today's master)

<!-- Bug, Feature, Enhancement, Etc -->
#### Type: bug

<!-- One of following:
    Critical - System crash, application panic.
    High - The main functionality of the application does not work, API breakage, repo format breakage, etc.
    Medium - A non-essential functionality does not work, performance issues, etc.
    Low - An optional functionality does not work.
    Very Low - Translation or documentation mistake. Something that really does not matter much but should be noticed for a future release. -->
#### Severity: low

#### Description:

```
> cd $GOPATH/src/github.com/ipfs/go-ipfs
> ipfs add -r .
added QmNxgZ2uqh12m9Y2q7gRHQa8hCgXmwf3ZBu2mDz3L6ocUb go-ipfs/CHANGELOG.md
[snip]
 135.70 KB / ? [------------------------------------=---------------------------------------------------------------------------------] ^C
Error: context canceled
panic: context canceled

goroutine 66 [running]:
gx/ipfs/QmamUWYjFeYYzFDFPTvnmGkozJigsoDWUA4zoifTRFTnwK/go-ipfs-cmds.(*chanResponseEmitter).SetError(0xc4201dbda0, 0x12c3420, 0xc420046a10, 0x4)
	/home/user/workspace/gopath/src/gx/ipfs/QmamUWYjFeYYzFDFPTvnmGkozJigsoDWUA4zoifTRFTnwK/go-ipfs-cmds/chan.go:186 +0x121
main.callCommand.func1(0xc4201d0b90, 0x1cb55e0, 0xc4201dbda0, 0x1cb6a40, 0xc4202f0000)
	/home/user/protocol/ipfs/go-ipfs/cmd/ipfs/main.go:369 +0xa6
created by main.callCommand
	/home/user/protocol/ipfs/go-ipfs/cmd/ipfs/main.go:366 +0x9e8
```

--- a/cmd/ipfs/main.go
+++ b/cmd/ipfs/main.go
@@ -364,9 +364,12 @@ func callCommand(ctx context.Context, req cmds.Request, root *cmds.Command, cmd 
 		}
 
 		go func() {
-			err = cmds.Copy(re, res)
+			err := cmds.Copy(re, res)
 			if err != nil {
-				re.SetError(err, cmdkit.ErrNormal|cmdkit.ErrFatal)
+				err = re.Emit(cmdkit.Error{err.Error(), cmdkit.ErrNormal | cmdkit.ErrFatal})
+				if err != nil {
+					log.Error(err)
+				}
 			}
 		}()
 	} else {

